### ISSUE TITLE
assignment of dunder methods in class reports incompatible `Self` type.

### ISSUE THREAD

**Issue by kennipj** [microsoft/pyright#2756]
It's common to directly assign dunder methods like `__repr__` to other dunder methods with the same function signature, when the same behaviour is desired, rather than defining a new function. However, pyright will now report `type "Self@object" cannot be assigned to type "Self@SomeClass"` in such cases.

Example:
```py
class SomeClass:
    def __str__(self) -> str:
        ...

    __repr__ = __str__ # Parameter 1: type "Self@object" cannot be assigned to type "Self@SomeClass"
```

I would expect no error to be reported in this instance

Tested on pyright 1.1.199

**Comment by erictraut** [comment]
I've added some special-case handling to accommodate this case.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.200, which I just published. It will also be included in the next release of pylance.
